netsim/ansible/tasks/fortinet.fortios.fortios/initial.yml

```diff
@@ -51,88 +51,6 @@
 
-- name: Configure global attributes
-  fortinet.fortios.fortios_system_global:
-    vdom: "{{ netlab_vdom }}"
-    system_global:
-      hostname: '{{ inventory_hostname.replace("_","-") }}'
-
-- name: Turn off LLDP on management interface
-  fortinet.fortios.fortios_system_interface:
-    vdom: "{{ vdom }}"
-    state: "present"
-    system_interface:
-      interface: "{{ mgmt_if }}"
-      lldp_reception: "disable"
-      lldp_transmission: "disable"
-      name: "{{ mgmt_if }}"
-
-- name: Configure loopback interface
-  fortinet.fortios.fortios_system_interface:
-    vdom: "{{ netlab_vdom }}"
-    state: "present"
-    system_interface:
-      interface: "loopback0"
-      ip: "{{ loopback.ipv4 | default(omit) }}"
-      name: "loopback0"
-      type: "loopback"
-      vdom: "{{ netlab_vdom }}"
-      allowaccess: "ping"
-  when: loopback is defined
-
-- name: Configure loopback ipv6 address
-  fortinet.fortios.fortios_system_interface:
-    vdom: "{{ netlab_vdom }}"
-    state: "present"
-    system_interface:
-      interface: "loopback0"
-      ipv6:
-        ip6_address: "{{ loopback.ipv6 }}"
-        ip6_mode: "static"
-      name: "loopback0"
-      vdom: "{{ netlab_vdom }}"
-  when: loopback.ipv6 is defined
-
-- name: Configure physical interfaces
-  fortinet.fortios.fortios_system_interface:
-    vdom: "{{ netlab_vdom }}"
-    state: "present"
-    system_interface:
-      alias: '{{ interface.name.replace(">","-") | default(omit) }}'
-      estimated_upstream_bandwidth: "{{ interface.bandwidth | default(omit) }}"
-      estimated_downstream_bandwidth: "{{ interface.bandwidth | default(omit) }}"
-      interface: "{{ interface.ifname }}"
-      ip: "{{ interface.ipv4 | default(omit) }}"
-      lldp_reception: "enable"
-      lldp_transmission: "enable"
-      mode: "static"
-      mtu: "{{ interface.mtu | default(mtu) }}"
-      mtu_override: "enable"
-      name: "{{ interface.ifname }}"
-      macaddr: "52:dc:ca:fe:{{ id }}:{{ interface.ifindex }}"
-      type: "physical"
-      vdom: "{{ netlab_vdom }}"
-      allowaccess: "ping"
-  with_items: "{{ interfaces }}"
-  when: interface.ifname != "{{ mgmt_if }}"
-  loop_control:
-    loop_var: interface
-
-- name: Configure interface ipv6 addresses
-  fortinet.fortios.fortios_system_interface:
-    vdom: "{{ netlab_vdom }}"
-    state: "present"
-    system_interface:
-      interface: "{{ interface.ifname }}"
-      ipv6:
-        ip6_address: "{{ interface.ipv6 }}"
-        ip6_mode: "static"
-        ip6_allowaccess: "ping"
-        ip6_send_adv: "enable"
-        ra_send_mtu: "enable"
-      name: "{{ interface.ifname }}"
-      vdom: "{{ netlab_vdom }}"
-  with_items: "{{ interfaces }}"
-  when: interface.ipv6 is defined
-  loop_control:
-    loop_var: interface
+- name: Deploy initial configuration from template
+  include_tasks: tasks/deploy-config/fortinet.fortios.fortios.yml
+  when: config_template is defined and config_template is string and config_template != ''
```

Comment:
Considering the initial template must be defined, I think we can skip the `when` condition. Also, the `deploy-config` task would take care of that anyway.